mlrun/datastore/sources.py

```diff
@@ -361,13 +370,14 @@ def to_step(
     ):
         import storey
 
-        attributes = self.attributes or {}
+        attributes = self.attributes.copy() or {}
```

Comment:
Why deepcopy in some places and copy in others?

Reply:
Also, it's not clear what's the point of falling back on `{}`? It seems redundant here.

Reply:
You are right, in the builder method of the base class if attributes is None, it sets it to empty dict.

Reply:
It cannot be `None` anyway because the constructor already replaces `None` with `{}`.